admiral/pkg/clusters/virtualservice_routing.go

```diff
@@ -883,21 +905,61 @@ func getDestinationsForGTPDNSPrefixes(
 			continue
 		}
 
+		var newDNSPrefixedSNIHost, routeHost string
+
 		for _, policy := range globalTrafficPolicy.Spec.Policy {
-			if policy.DnsPrefix == common.Default || policy.DnsPrefix == env {
+			weights := make(map[string]int32)
+			var remoteRD *vsrouting.RouteDestination
+			if policy.Target != nil {
+				for _, target := range policy.Target {
+					weights[target.Region] = target.Weight
+				}
+			}
+
+			if !updateWeights && len(weights) == 0 && (policy.DnsPrefix == common.Default || policy.DnsPrefix == env) {
 				continue
+			} else if policy.DnsPrefix == common.Default || policy.DnsPrefix == env {
+				routeHost = hostWithoutSNIPrefix
+			} else {
+				routeHost = policy.DnsPrefix + common.Sep + hostWithoutSNIPrefix
 			}
-			newDNSPrefixedSNIHost, err := generateSNIHost(policy.DnsPrefix + common.Sep + hostWithoutSNIPrefix)
+
+			newDNSPrefixedSNIHost, err = generateSNIHost(routeHost)
 			if err != nil {
 				return nil, err
 			}
+
 			newRD, err := copyRouteDestinations(routeDestinations)
 			if err != nil {
 				return nil, err
 			}
+
+			if len(weights) > 0 && updateWeights {
+				for _, rd := range newRD {
+					if strings.HasSuffix(rd.Destination.Host, "svc.cluster.local") {
```

Comment:
nitpick,  can you please use `common.DotLocalDomainSuffix`?

Reply:
updated it